components/scream/src/mct_coupling/tests/scorpio_tests.cpp

```diff
@@ -107,30 +218,153 @@ TEST_CASE("scorpio_interface_output", "") {
+  get_variable(outfilename,"index_1d","test value for 1d field",1,vec_x, PIO_INT,"x-int_in");
+  get_variable(outfilename,"index_2d","test value for 2d field",2,vec_xy, PIO_INT,"xy-int_in");
+  get_variable(outfilename,"index_3d","test value for 3d field",3,vec_xyz, PIO_INT,"xyz-int_in");
+  /* 
+   * Setup PIO configuation for reading data.  This is usually a 3 step process:
+   * 1. Determine the number of indices, start and stop location for this rank.
+   * 2. Create a vector of the indices based on the info from step 1. 
+   *    We need to do this as a second step since we don't know the length of this vector ahead of time.
+   * 3. Assign the global indices to the dof vector.
+   * We already did all of this setting things up for writing output, so here we just reuse the same dof vectors.
+   */
+  set_dof(outfilename,"time",0,0);
+  set_dof(outfilename,"x",dof_x[0],x_dof.data());
+  set_dof(outfilename,"y",dof_y[0],y_dof.data());
+  set_dof(outfilename,"z",dof_z[0],z_dof.data());
+  set_dof(outfilename,"index_1d",dof_test_1d[0],test1d_dof.data());
+  set_dof(outfilename,"data_1d",dof_test_1d[0],test1d_dof.data());
+  set_dof(outfilename,"index_2d",dof_test_2d[0],test2d_dof.data());
+  set_dof(outfilename,"data_2d",dof_test_2d[0],test2d_dof.data());
+  set_dof(outfilename,"index_3d",dof_test_3d[0],test3d_dof.data());
+  set_dof(outfilename,"data_3d",dof_test_3d[0],test3d_dof.data());
+  // Now that the variables have been gathered and the dof set for each one, set the iodesc decomposition from PIO.
+  set_decomp(outfilename);
+  /* Read input data for x, y and z, and compare with expected value.  Report mismatches as an error. */
+  grid_read_data_array(outfilename,"x",dof_x[0],ekat::util::data(x_data)+xstart);
+  grid_read_data_array(outfilename,"y",dof_y[0],ekat::util::data(y_data)+ystart);
+  grid_read_data_array(outfilename,"z",dof_z[0],ekat::util::data(z_data)+zstart);
+  nerr = 0;
+  for (Int ii=xstart;ii<=xstop;ii++) {
+    if (x_data[ii] != x_i(ii,x_data.size())) { ++nerr;}
+  }
+  REQUIRE(nerr==0);
+  nerr = 0;
+  for (Int jj=ystart;jj<=ystop;jj++) {
+    if (y_data[jj] != y_i(jj,y_data.size())) { ++nerr;}
+  }
+  REQUIRE(nerr==0);
+  nerr = 0;
+  for (Int kk=zstart;kk<=zstop;kk++) {
+    if (z_data[kk] != z_i(kk,z_data.size())) { ++nerr; std::printf("   - (%d) %f vs. %f\n",kk,z_data[kk],z_i(kk,z_data.size()));}
+  }
+  REQUIRE(nerr==0);
+  /* Now read and compare the field data for each of the timestep */
+  nerr = 0;
+  for (int tt=0;tt<3;tt++) {
+    pio_update_time(outfilename,-999.0);
+    grid_read_data_array(outfilename,"index_1d",dof_test_1d[0],ekat::util::data(test_index_1d)+test_1d_start);
+    grid_read_data_array(outfilename,"data_1d", dof_test_1d[0],ekat::util::data(test_data_1d) +test_1d_start);
+    nerr = 0;
+    for (int ii=0,ind=test_1d_start;ind<test_1d_stop;ii++,ind++) {
+      if (*(ekat::util::data(test_data_1d) + ind)  != f_x(x_data[ind],tt*dt)) {++nerr;}
```

Comment:
People might have a different opinion here (I know P3 went this route already), but I find the `nerr` thing a bit pointless. Why not simply do

```
for(...) {
  REQUIRE ((ekat::util::data(test_data_1d) + ind) == f_x(x_data[ind],tt*dt));
}
```

It keeps the code leaner, and we kill the simulation right away.

Reply:
Normally I would agree, but when it comes to input/output I think it is helpful to know how many failed.  Things like "only half" or "only one" or "all" failed really helps identify what might be going wrong.  For example, if it's only half that fail maybe the issue is with the setting of dof's, and not in the actual reading of data.